include/cuco/detail/static_map/static_map.inl

```diff
@@ -318,11 +320,47 @@ void static_map<Key, T, Extent, Scope, KeyEqual, ProbingScheme, Allocator, Stora
   auto const num = cuco::detail::distance(first, last);
   if (num == 0) { return; }
 
-  auto const grid_size = cuco::detail::grid_size(num, cg_size);
-
-  static_map_ns::detail::insert_or_apply<cg_size, cuco::detail::default_block_size()>
-    <<<grid_size, cuco::detail::default_block_size(), 0, stream.get()>>>(
-      first, num, op, ref(op::insert_or_apply));
+  int32_t constexpr shmem_block_size = 1024;
+  int32_t const default_grid_size    = cuco::detail::grid_size(num, cg_size);
+
+  int32_t constexpr cardinality_threshold   = shmem_block_size;
+  int32_t constexpr shared_map_num_elements = cardinality_threshold + shmem_block_size;
+  float constexpr load_factor               = 0.7;
+  int32_t constexpr shared_map_size         = (1.0 / load_factor) * shared_map_num_elements;
+
+  using extent_type            = cuco::extent<int32_t, shared_map_size>;
+  using shared_map_type        = cuco::static_map<Key,
+                                                  T,
+                                                  extent_type,
+                                                  cuda::thread_scope_block,
+                                                  KeyEqual,
+                                                  ProbingScheme,
+                                                  Allocator,
+                                                  cuco::storage<1>>;
+  using shared_map_ref_type    = typename shared_map_type::ref_type<>;
+  auto constexpr window_extent = cuco::make_window_extent<shared_map_ref_type>(extent_type{});
+
+  using ref_type = decltype(ref(op::insert_or_apply));
+
+  auto insert_or_apply_shmem_fn_ptr = static_map_ns::detail::
+    insert_or_apply_shmem<cg_size, shmem_block_size, shared_map_ref_type, InputIt, Op, ref_type>;
+
+  int32_t const max_op_grid_size =
+    cuco::detail::max_occupancy_grid_size(shmem_block_size, insert_or_apply_shmem_fn_ptr, 0);
+
+  auto const shmem_grid_size      = std::min(default_grid_size, max_op_grid_size);
```

Comment:
If you go with `std::min` then please include `<algorithm>`. Alternatively you can just use the builtin `min` function or a ternary expression.